docs/design/scheduler_extender.md

```diff
@@ -0,0 +1,100 @@
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Scheduler extender
+
+Kubernetes scheduler schedules based on resources managed by Kubernetes. Scheduler extender is a way to extend the scheduling capability of k8s for resources that are not managed by k8s. A couple of example use cases are storage affinity and network availability.
+
+When scheduling a pod, the extender allows an external process to filter and prioritize nodes. Two separate http/https calls are issued through the extender for "filter" and "prioritize" actions. Extender's configuration is part of the scheduler policy file. The configuration specifies how to reach the extender and whether to use http or https.
+
+```
+// Holds the parameters used to communicate with the extender. If a verb is unspecified/empty,
+// it is assumed that the extender chose not to provide that extension.
+type ExtenderConfig struct {
+	// URLPrefix at which the extender is available
+	URLPrefix string `json:"urlPrefix"`
+	// Verb for the filter call, empty if not supported
+	FilterVerb string `json:"filterVerb,omitempty"`
+	// Verb for the prioritize call, empty if not supported
+	PrioritizeVerb string `json:"prioritizeVerb,omitempty"`
+	// EnableHttps specifies whether https should be used to communicate with the extender
+	EnableHttps bool `json:"enableHttps,omitempty"`
+	// TLSConfig specifies the transport layer security config
+	TLSConfig *client.TLSClientConfig `json:"tlsConfig,omitempty"`
+	// HTTPTimeout specifies the timeout duration for a call to the extender
+	HTTPTimeout time.Duration `json:"httpTimeout,omitempty"`
+}
+```
+
+A sample scheduler policy file with extender configuration:
+
+```
+{
+  "predicates": [
+    {
+      "name": "HostName"
+    },
+    {
+      "name": "MatchNodeSelector"
+    },
+    {
+      "name": "PodFitsResources"
+    }
+  ],
+  "priorities": [
+    {
+      "name": "LeastRequestedPriority",
+      "weight": 1
+    }
+  ],
+  "extenders": [
+    {
+      "urlPrefix": "http://127.0.0.1:12345/api/scheduler",
+      "filterVerb": "filter",
+      "enableHttps": false
+    }
+  ]
+}
+```
+
+Arguments passed to the extender are the set of nodes filtered through the k8s predicates and the pod.
+
+```
+// ExtenderArgs represents the arguments needed by the extender to filter/prioritize
+// nodes for a pod.
+type ExtenderArgs struct {
+	Pod   api.Pod      `json:"pod"`
+	Nodes api.NodeList `json:"nodes"`
+}
+```
+
+The "filter" call may further prune the set of nodes based on its predicates. Scores returned by the "prioritize" call are added to the k8s scores (computed through its priority functions) and used for final host selection.
```

Comment:
s/further prune/prune/

Also, add the struct that represents what is returned.